assets/js/modules/analytics/util/report-normalization.js

```diff
@@ -51,3 +53,26 @@ const normalizeMetrics = ( metrics ) => {
 		.filter( ( metric ) => isPlainObject( metric ) )
 	;
 };
+
+const normalizeDimensions = ( dimensions ) => {
+	return castArray( dimensions )
+		.map(
+			( dimension ) => typeof dimension === 'string'
+				? { name: dimension }
+				: dimension
+		)
+		.filter( ( dimension ) => isPlainObject( dimension ) )
+	;
+};
+
+const normalizeDimensionFilters = ( dimensionFilters ) => {
+	const normalizedFilters = {};
+	if ( isPlainObject( dimensionFilters ) ) {
+		for ( const [ key, value ] of Object.entries( dimensionFilters ) ) {
+			if ( typeof value === 'string' ) {
+				normalizedFilters[ key ] = value;
+			}
+		}
+	}
+	return normalizedFilters;
+};
```

Comment:
I don't think we need a normalization function for dimension filters. The reason they exist for dimensions and metrics is that those arguments support multiple different ways of being specified (e.g. string, array of strings, array of objects). For dimension filters though, for now we only support a map of scalar values - so let's remove `normalizeDimensionFilters`. We only need validation here.